### DISCUSSION TITLE
MOOSE build fail on INL HPC

### DISCUSSION THREAD

**Discussion by ver2king** [idaholab/moose#28294] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

I tempted to build MOOSE on INL HPC for one of my projects, and I strictly followed the procedure outlined here: 

https://mooseframework.inl.gov/getting_started/installation/inl_hpc_install_moose.html

Everything was fine until these commands were executed :

cd ~/cluster_name/projects/moose/test
make -j 6

The generated errors are screen captured here: 

![3853e59a-be61-4063-8608-d77719a3d913](https://github.com/user-attachments/assets/44e5fa56-c205-46fb-8320-a09819ec36c5)

Please assist me further with this error. I have tried this via Sawtooth2 shell terminal, and launched it from INL HPC on demand via my employee credentials.

**Comment by milljm** [comment]
This annotation is pretty important, and may be the cause for the failure.

<img width="979" alt="Screenshot 2024-08-02 at 11 19 41 AM" src="https://github.com/user-attachments/assets/c7a38980-f0c4-4d7e-b4b0-df698588c63d">


Basically, if you ran `make` in `~/cluster_name/projects/moose` using any other machine other than sawtooth (the one you are using now), the error you are receiving may occur. Which is why we instruct users to separated their projects directory, labeled by way of HPC clusters available:
```bash
ssh sawtooth1 (or 2)
cd ~/sawtooth/projects/moose
make

ssh lemhi1 (or 2)
cd ~/lemhi/projects/moose
make

# Bitterroot is broken at the moment, but the idea is the same
ssh bitterroot1 (or 2)
cd ~/bitterroot/projects/moose
make
```
and etc.

In order to fix this, you will need to clean out your current repository, and build anew (add any untracked files before doing so, or they will be lost. If you have made no changes to MOOSe, you can ignore this comment):
```bash
module purge
module load use.moose moose-dev
cd ~/cluster_name/projects/moose
git clean -xfd
git submodule foreach --recursive git clean -xfd
```

Once cleaned, you should be able to run `make`:
```bash
cd test
make -j 6
```

**Reply by milljm** [reply]
I forgot to mention:

It might also be necessary to rebuild PETSc, libMesh, and WASP, for the same reasons above.

**Comment by jakson121** [comment]
I encountered the same issue as you. Did you manage to solve it?
![image](https://github.com/user-attachments/assets/1998b22d-3dec-41d0-a5ad-a08ebee0d557)

**Reply by ver2king** [reply]
Hi @jakson121,

I posted this question in the context of MOOSE compiled via INL's HPC. IF you compiled MOOSE under a similar context, please let me know and we ma y figure things out.

**Reply by milljm** [reply]
@jakson121 's originating issue: https://github.com/idaholab/moose/discussions/28345

**Reply by milljm** [reply]
@ver2king, were you able to resolve this?

When I build on sawtooth2, using `module load use.moose moose-dev`, I am not running into an issue.

**Reply by ver2king** [reply]
@milljm , I were able to solve this. Before using `module load use.moose moose-dev`, ensure to enter the correct cluster via `ssh` (for example: `ssh sawtooth1`). Also, ensure to re-build all petsc, libmesh and wasp. I tried on both sawtooth1 and sawtooth2 and have seen no problem.
